corehq/apps/app_manager/models.py

```diff
@@ -1748,6 +1752,17 @@ class CustomIcon(DocumentSchema):
     text = DictProperty(str)
     xpath = StringProperty()
 
+    def _type(self):
+        return CUSTOM_ICON_TYPE_XPATH if self.xpath else CUSTOM_ICON_TYPE_TEXT
+
+    @property
+    def is_text(self):
+        return self._type() == CUSTOM_ICON_TYPE_TEXT
+
+    @property
+    def is_xpath(self):
+        return self._type() == CUSTOM_ICON_TYPE_XPATH
+
```

Comment:
Nit: Looks like `_type()` is only used here. You could probably simplify this a little:

```suggestion
    @property
    def is_text(self):
        return not self.is_xpath

    @property
    def is_xpath(self):
        return bool(self.xpath)
```